internal/hooks/v0hooks/manager.go

```diff
@@ -266,192 +185,27 @@ func (o *Manager) runHook(
-
-		} else if err != nil {
-			if terr, ok := err.(net.Error); ok && terr.Timeout() || i < DefaultHTTPHookRetries-1 {
-				hookLog.Errorf("Request timed out for attempt %d with err %s", i, err)
-				time.Sleep(HTTPHookBackoffDuration)
-				continue
-			} else if i == DefaultHTTPHookRetries-1 {
-				return nil, apierrors.NewUnprocessableEntityError(apierrors.ErrorCodeHookTimeoutAfterRetry, "Failed to reach hook after maximum retries")
-			} else {
-				return nil, apierrors.NewInternalServerError("Failed to trigger auth hook, error making HTTP request").WithInternalError(err)
-			}
-		}
-
-		defer rsp.Body.Close()
+func checkError(
+	hookOutput HookOutput,
+) error {
+	if hookOutput.IsError() {
+		he := hookOutput.GetHookError()
+		httpCode := he.HTTPCode
 
-		switch rsp.StatusCode {
-		case http.StatusOK, http.StatusNoContent, http.StatusAccepted:
-			// Header.Get is case insensitive
-			contentType := rsp.Header.Get("Content-Type")
-			if contentType == "" {
-				return nil, apierrors.NewBadRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, "Invalid Content-Type: Missing Content-Type header")
-			}
-			mediaType, _, err := mime.ParseMediaType(contentType)
-			if err != nil {
-				return nil, apierrors.NewBadRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, fmt.Sprintf("Invalid Content-Type header: %s", err.Error()))
-			}
-			if mediaType != "application/json" {
-				return nil, apierrors.NewBadRequestError(apierrors.ErrorCodeHookPayloadInvalidContentType, "Invalid JSON response. Received content-type: "+contentType)
-			}
-			if rsp.Body == nil {
-				return nil, nil
-			}
-			limitedReader := io.LimitedReader{R: rsp.Body, N: PayloadLimit}
-			body, err := io.ReadAll(&limitedReader)
-			if err != nil {
-				return nil, err
-			}
-			if limitedReader.N <= 0 {
-				// check if the response body still has excess bytes to be read
-				if n, _ := rsp.Body.Read(make([]byte, 1)); n > 0 {
-					return nil, apierrors.NewUnprocessableEntityError(apierrors.ErrorCodeHookPayloadOverSizeLimit, fmt.Sprintf("Payload size exceeded size limit of %d bytes", PayloadLimit))
-				}
-			}
-			return body, nil
-		case http.StatusTooManyRequests, http.StatusServiceUnavailable:
-			retryAfterHeader := rsp.Header.Get("retry-after")
-			// Check for truthy values to allow for flexibility to switch to time duration
-			if retryAfterHeader != "" {
-				continue
-			}
-			return nil, apierrors.NewInternalServerError("Service currently unavailable due to hook")
-		case http.StatusBadRequest:
-			return nil, apierrors.NewInternalServerError("Invalid payload sent to hook")
-		case http.StatusUnauthorized:
-			return nil, apierrors.NewInternalServerError("Hook requires authorization token")
-		default:
-			return nil, apierrors.NewInternalServerError("Unexpected status code returned from hook: %d", rsp.StatusCode)
+		if httpCode == 0 {
```

Comment:
guess this happens for pg hooks? would be good to include a comment here on when a 0 might be returned

Reply:
All outputs include AuthHookError type under `error` field, so you can construct a JSON response in pgfunc that includes it just like the HTTP version.